### ISSUE TITLE
Creating a CentOS bootable image fails

### ISSUE THREAD

**Issue by stejoo** [systemd/mkosi#356]
When I try to create a bootable CentOS image the build fails because it is unable to find package `systemd-udev`. I have checked CentOS repositories and this package does not seem to exist within CentOS base, updates or extras repositories. It does, of course, exist on Fedora and I can find a systemd-udev RPM for EL7 online after some searching. But still it's not in the base repositories... So depending on it puzzles me.

This prevents me from creating a bootable CentOS image at this time.
I'll try to fix it on my end of course.
What exactly to we need of `systemd-udev` to make an image bootable? I can see a lot of the files provided by this package on Fedora are also present in the `systemd` package on CentOS. I will dig deeper when I find some time.

In case you guys got any ideas on what is required here or how to resolve this issue I would appreciate it. And of course if it's me doing it wrong, please let me know :)

**Comment by purpleidea** [comment]
I can confirm I get this issue trying to build a CentOS-7 image. I get:
```
No match for argument: systemd-udev
```

Bug is here:

https://github.com/systemd/mkosi/blob/c7833b9b21e83e9f5c502584bb029bddd831302a/mkosi#L1633

I don't know the root cause or what the correct package is, or why this rotted. Maybe someone else does?

Workaround to make things build (but not necessarily get a working image) is:

```
[Output]
#Bootable must be no or else systemd-udev package will be installed (naming bug)
Bootable=no
```

More logs:
```
Traceback (most recent call last):
  File "/usr/bin/mkosi", line 4644, in <module>
    main()
  File "/usr/bin/mkosi", line 4640, in main
    run_verb(a)
  File "/usr/bin/mkosi", line 4615, in run_verb
    build_stuff(args)
  File "/usr/bin/mkosi", line 4430, in build_stuff
    raw, tar, root_hash = build_image(args, workspace, do_run_build_script=True)
  File "/usr/bin/mkosi", line 4258, in build_image
    do_run_build_script=do_run_build_script, cached=cached)
  File "/usr/bin/mkosi", line 2048, in install_distribution
    install[args.distribution](args, workspace, do_run_build_script)
  File "/usr/lib64/python3.7/contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "/usr/bin/mkosi", line 1644, in install_centos
    config_file)
  File "/usr/bin/mkosi", line 1561, in invoke_dnf_or_yum
    invoke_dnf(args, workspace, repositories, packages, config_file)
  File "/usr/bin/mkosi", line 1350, in invoke_dnf
    run(cmdline, check=True)
  File "/usr/bin/mkosi", line 76, in run
    return subprocess.run(cmdline, **kwargs)
  File "/usr/lib64/python3.7/subprocess.py", line 487, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['dnf', '-y', '--config=/var/tmp/mkosi-siiu0v3q/yum.conf', '--best', '--allowerasing', '--releasever=7', '--installroot=/var/tmp/mkosi-siiu0v3q/root', '--disablerepo=*', '--enablerepo=base', '--enablerepo=updates', '--enablerepo=extras', '--enablerepo=centosplus', '--setopt=keepcache=1', '--setopt=install_weak_deps=0', '--nodocs', 'install', 'centos-release', 'dnf', 'make', 'git', 'tree', 'kernel', 'systemd-udev', 'binutils', 'btrfs-progs']' returned non-zero exit status 1.

```

HTH

**Comment by poettering** [comment]
I guess fixed by #355. Closing.

**Comment by stejoo** [comment]
No, I made this a separate issue because this was not fixed by my pull request (#355). Please re-open this.

The CentOS flavor only works when building a non-bootable image. When trying to create a bootable image yum fails complaining about being unable to find the `systemd-udev` package.

**Comment by poettering** [comment]
Oops. Sorry

**Comment by mchugh19** [comment]
From a Debian 10 host, running 
` mkosi -d centos -O /home/centos7 -r 7 -b` results in 
```
‣ Unmounting image complete.
‣ Detaching image file...
‣ Detaching image file complete.
Traceback (most recent call last):
  File "/usr/bin/mkosi", line 4872, in <module>
    main()
  File "/usr/bin/mkosi", line 4868, in main
    run_verb(a)
  File "/usr/bin/mkosi", line 4843, in run_verb
    build_stuff(args)
  File "/usr/bin/mkosi", line 4667, in build_stuff
    raw, tar, root_hash = build_image(args, workspace, do_run_build_script=False, cleanup=True)
  File "/usr/bin/mkosi", line 4479, in build_image
    install_boot_loader(args, workspace.name, loopdev, cached)
  File "/usr/bin/mkosi", line 2381, in install_boot_loader
    os.path.join(workspace, "root", "efi/EFI/systemd/systemd-bootx64.efi"))
  File "/usr/lib/python3.7/shutil.py", line 120, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: '/var/tmp/mkosi-1dp2ocys/root/usr/lib/systemd/boot/efi/systemd-bootx64.efi'

```

**Comment by DaanDeMeyer** [comment]
This should be fixed by #434.
